funding/nonces.go

```diff
@@ -0,0 +1,214 @@
+			m.annSigNonces[chanID] = annSigNonces{
+				received: &receivedNonces{
+					btc:  n.Btc,
+					node: n.Node,
+				},
+			}
+		}
+	})
+
+	return returnErr
+}
+
+func (m *nonceManager) completed(chanID lnwire.ChannelID) error {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	delete(m.annSigNonces, chanID)
+
+	return m.store.DeleteAnnouncementNonces(chanID)
+}
+
+func (m *nonceManager) getReceivedNonces(chanID lnwire.ChannelID) (
+	[musig2.PubNonceSize]byte, [musig2.PubNonceSize]byte, bool) {
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	aggNonces, ok := m.annSigNonces[chanID]
+	if !ok || aggNonces.received == nil {
+		return [musig2.PubNonceSize]byte{}, [musig2.PubNonceSize]byte{},
+			false
+	}
+
+	return aggNonces.received.btc, aggNonces.received.node, true
+}
+
+func (m *nonceManager) receivedNonces(chanID lnwire.ChannelID, node,
+	btc [musig2.PubNonceSize]byte, revProducer shachain.Producer,
+	multiSigBTCKey *btcec.PublicKey) error {
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	sendNonces, err := m.getNoncesToSendUnsafe(
+		chanID, revProducer, multiSigBTCKey,
+	)
+	if err != nil {
+		return err
+	}
+
+	announcementSendNonces, ok := m.annSigNonces[chanID]
+	if !ok {
+		m.annSigNonces[chanID] = annSigNonces{
+			sent: sendNonces,
+		}
+		announcementSendNonces = m.annSigNonces[chanID]
+	}
+
+	announcementSendNonces.received = &receivedNonces{
+		btc:  btc,
+		node: node,
+	}
+
+	m.annSigNonces[chanID] = announcementSendNonces
+
+	return m.store.SaveAnnouncementNonces(
+		chanID, &channeldb.AnnouncementNonces{
+			Node: node,
+			Btc:  btc,
+		},
+	)
+}
+
+func (m *nonceManager) getNoncesToSend(chanID lnwire.ChannelID,
+	revProducer shachain.Producer, multiSigBTCKey *btcec.PublicKey) (
+	*musig2.Nonces, *musig2.Nonces, error) {
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	nonces, err := m.getNoncesToSendUnsafe(
+		chanID, revProducer, multiSigBTCKey,
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return nonces.btc, nonces.node, nil
+}
+
+func (m *nonceManager) getNoncesToSendUnsafe(chanID lnwire.ChannelID,
+	revProducer shachain.Producer, multiSigBTCKey *btcec.PublicKey) (
+	*sentNonces, error) {
+
+	announceNonces, ok := m.annSigNonces[chanID]
+	if ok && announceNonces.sent != nil {
+		return announceNonces.sent, nil
+	}
+
+	sent, err := m.genAnnouncementNonces(revProducer, multiSigBTCKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if !ok {
+		m.annSigNonces[chanID] = annSigNonces{sent: sent}
+	} else {
+		announceNonces.sent = sent
+	}
+
+	announceNonces = m.annSigNonces[chanID]
+
+	return sent, nil
+}
+
+func (m *nonceManager) genAnnouncementNonces(revProducer shachain.Producer,
+	multiSigBTCKey *btcec.PublicKey) (*sentNonces, error) {
+
+	musig2ShaChain, err := channeldb.DeriveMusig2Shachain(
+		revProducer,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to generate musig "+
+			"announcement nonces: %v", err)
+	}
+
+	// TODO: replace with unique derivation.
+	btcNonce, err := channeldb.NewMusigVerificationNonce(
```

Comment:
So just fully randomized? 

If we're using the `shachain` based derivation, then we actually never need to write them to disk.